README.md

```diff
@@ -130,6 +133,135 @@ For Facebook Messenger, the approach is the same, but the JSON details differ:
+      }
+    ]
+  }
+}
+```
+
+**Slack**
+The generic response array is translated to a *single* message in Slack comprising of text alongwith attachment data. For the above example, the equivalent Slack message after translation is as follows:
+
+```json
+{
+  "text": "Here are your nearest stores.",
+  "attachments": [
+    {
+      "image_url": "http://...",
+      "pretext": "Some description for the image",
+      "title": "Image title"
+    },
+    {
+      "actions": [
+        {
+          "name": "Location 1",
+          "text": "Location 1",
+          "type": "button",
+          "value": "Location 1"
+        },
+        {
+          "name": "Location 2",
+          "text": "Location 2",
+          "type": "button",
+          "value": "Location 2"
+        },
+        {
+          "name": "Location 3",
+          "text": "Location 3",
+          "type": "button",
+          "value": "Location 3"
+        }
+      ],
+      "callback_id": "Click on one of the following",
+      "text": "Click on one of the following"
+    }
+  ],
+  "channel": "DXXXXXXXX",
+  "ts": "XXXXXXXXXX.XXXXXX"
+}
+}
+```
+**Facebook**
+The generic response array is translated to a *list of Facebook messages payloads* comprising of text along with attachment data. For the above example, the equivalent message list after translation is as follows:
+
+```json
+[
+  {
+    "text": "Here are your nearest stores."
+  },
+  {
+    "attachment": {
+      "type": "image",
+      "payload": {
+        "url": "http://..."
+      }
+    }
+  },
+  {
+    "text": "Click on one of the following",
+    "quick_replies": [
+      {
+        "content_type": "text",
+        "title": "Location 1",
+        "payload": "Location 1"
+      },
+      {
+        "content_type": "text",
+        "title": "Location 2",
+        "payload": "Location 2"
+      },
+      {
+        "content_type": "text",
+        "title": "Location 3",
+        "payload": "Location 3"
+      }
+    ]
+  }
+]
+```
+In case Facebook is the channel in question, the output of `normalize-conversation-for-<channel>` is sent to `multiple_post` which splits the array into individual Facebook messages and invokes a `sub-pipeline` comprising of `starter-code/post-normalize` and `facebook/post` functions. Each message is separately passed through this subpipeline. Hence, a multi-modal response from Conversation is broken down into individual messages and POSTed to Facebook messenger.
+
+Additionally, Facebook only allows upto 11 elements in a quick reply array. So, if the options list in the Conversation response has over 11 options, it's translated to the [Facebook generic template](https://developers.facebook.com/docs/messenger-platform/send-messages/template/generic) format with buttons in groups of three.
+
+*Note: Since multiple messages(coming from one Conversation response) will be fired off in quick succession, they may appear out of order if there's some delay in delivery or, Facebook servers are slow.*
```

Comment:
I think we might want to rephrase this, as far as our code I believe we have done everything we can to ensure they will arrive in order (found a bug on facebook's page where they recommend queuing and a community member recommends recursively sending messages and we are doing this). But yeah until facebook allows some sort of sequence number, timestamp, or array of messages I'm not sure we can theoretically get rid of this.